intake_stac/catalog.py

```diff
@@ -439,34 +419,59 @@ def _get_plot(self, item):
 
         return default_plot
 
-    def _get_driver(self, entry):
-
-        entry_type = entry.get('type')
+    def _get_driver(self, asset):
+        """
+        Assign intake driver for data I/O
+        """
+        entry_type = asset.media_type
 
         if entry_type in ['', 'null', None]:
 
-            suffix = os.path.splitext(entry['href'])[-1]
+            suffix = os.path.splitext(asset.media_type)[-1]
             if suffix in ['.nc', '.h5', '.hdf']:
-                entry['type'] = 'application/netcdf'
+                asset.media_type = 'application/netcdf'
                 warnings.warn(
-                    f'STAC Asset "type" missing, assigning {entry_type} based on href suffix {suffix}:\n{entry}'  # noqa: E501
+                    f'STAC Asset "type" missing, assigning {entry_type} based on href suffix {suffix}:\n{asset.media_type}'  # noqa: E501
                 )
             else:
-                entry['type'] = default_type
+                asset.media_type = default_type
                 warnings.warn(
-                    f'STAC Asset "type" missing, assuming default type={default_type}:\n{entry}'
+                    f'STAC Asset "type" missing, assuming default type={default_type}:\n{asset}'  # noqa: E501
                 )
-            entry_type = entry.get('type')
+            entry_type = asset.media_type
             print(entry_type)
 
         # if mimetype not registered try rasterio driver
         driver = drivers.get(entry_type, default_driver)
 
         return driver
 
-    def _get_args(self, entry, driver, stacked=False):
-        args = entry if stacked else {'urlpath': entry.get('href')}
+    def _get_args(self, asset, driver):
+        """
+        Optional keyword arguments to pass to intake driver
+        """
+        args = {'urlpath': asset.href}
         if driver in ['netcdf', 'rasterio', 'xarray_image']:
+            # NOTE: force using dask?
             args.update(chunks={})
 
         return args
+
+
+class CombinedAssets(LocalCatalogEntry):
+    """
+    Maps multiple STAC Item Assets to 1 Intake Catalog Entry
+    """
```

Comment:
I thought it was easier to understand a separate 'CombinedAssets' class instead of having the keyword stacked=False in the StacAsset class with logic around how it is constructed. Right now this is only relevant to the StacItem.stack_bands() method but also want a more obvious way to implement other things such as StacCatalog.stack_items() or mosaic()...

Reply:
yep. This seems like a fine idea. ping @TomAugspurger who's been thinking about these things lately.

Reply:
Yeah, IIUC this way sounds good.